cmdline/tool/test/unit/src/org/netbeans/modules/jackpot30/cmdline/MainTest.java

```diff
@@ -861,6 +861,33 @@ private void runSourceLevelMatches(String sourceLevel, String expectedOutput) th
                       sourceLevel);
     }
 
+    public void testParameterFile() throws Exception {
+        String golden =
+            "package test;\n" +
+            "public class Test {\n" +
+            "    private void test(java.util.Collection c) {\n" +
+            "        boolean b = c.isEmpty();\n" +
+            "    }\n" +
+            "}\n";
+
+        doRunCompiler(golden,
+                      null,
+                      null,
+                      "src/test/Test.java",
+                      "package test;\n" +
+                      "public class Test {\n" +
+                      "    private void test(java.util.Collection c) {\n" +
+                      "        boolean b = c.size() == 0;\n" +
+                      "    }\n" +
+                      "}\n",
+                      "parameters.txt",
+                      "--apply\n" +
+                      "--hint\n" +
+                      "Usage of .size() == 0\n",
+                      null,
+                      "@" + getWorkDirPath() + "/parameters.txt");
```

Comment:
Where is the `parameters.txt` file? Shouldn't it also be part of the commit?

Reply:
The tests here are a little ugly here - the file is created on the fly using the lines above:
                      "parameters.txt",
                      "--apply\n" +
                      "--hint\n" +
                      "Usage of .size() == 0\n"

(the parameters are basically:
-expected file context
-expected stdout
-expected stderr
-filename and file context
-null
-parameters sent to the tool
)